DeepSpeech.py

```diff
@@ -1169,6 +1213,10 @@ def _next_epoch(self):
                     # The current epoch should also have a validation part
                     self._epochs_running.append(Epoch(self._epoch, self._num_jobs_dev, set_name='dev', report=is_display_step))
 
+                    # State Machine flag for decision making regarding early stopping in next epoch function call
+                    self._dev = True
```

Comment:
I thought it would be better to align with the state mahine design rather than checking ever epoch. Hence, enabling the flag only when a validation epoch is enqueued